spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/PositionDeletesRowReader.java

```diff
@@ -90,6 +91,10 @@ protected CloseableIterator<InternalRow> open(PositionDeletesScanTask task) {
         ExpressionUtil.extractByIdInclusive(
             task.residual(), expectedSchema(), caseSensitive(), Ints.toArray(nonConstantFieldIds));
 
+    if (ContentFileUtil.isDV(task.file())) {
```

Comment:
Do we need `Iterable` or can we simply implement `Iterator` directly? It seems we create an iterable for the sake of getting an iterator for it.

Reply:
you're right, we can implement/return an `Iterator` directly here